system/infra-monitoring/templates/_prometheus-infra-collector.yaml.tpl

```diff
@@ -20,7 +20,7 @@
     regex: .*infra-collector.*
   - action: keep
     source_labels: [__meta_kubernetes_pod_container_port_number, __meta_kubernetes_pod_container_port_name, __meta_kubernetes_pod_annotation_prometheus_io_port]
-    regex: (9102;.*;.*)|(.*;metrics;.*)|(.*;.*;\d+)
+    regex: (9102;.*;.*)|(.*;metrics(-.*)?;.*)|(.*;.*;\d+)
```

Comment:
I suggest to use the same regex `regex: '(9102;.*;.*)|(.*;metrics(-.*)?;.*)|(\d+;.*;\d+)|(.*;.*;\d+)'` as in pod-monitor https://github.com/sapcc/helm-charts/blob/master/common/prometheus-server/templates/podmonitors/pod-sd.yaml#L43-L47.